OutOfSchool/OutOfSchool.BusinessLogic/Services/Database/WorkshopService.cs

```diff
@@ -1165,4 +1140,77 @@ await UpdateStatus(new()
             }).ConfigureAwait(false);
         }
     }
+
+    private async Task<Workshop> CheckDtoAndPrepareCreatedWorkshop(WorkshopCreateUpdateDto dto)
+    {
+        if (dto.MemberOfWorkshopId.HasValue && !await Exists((Guid)dto.MemberOfWorkshopId).ConfigureAwait(false))
+        {
+            var errorMessage = $"The main workshop (with id = {dto.MemberOfWorkshopId}) for the workshop being created was not found.";
+            throw new InvalidOperationException(errorMessage);
+        }
+
+        if (dto.MemberOfWorkshopId.HasValue && (await GetById((Guid)dto.MemberOfWorkshopId).ConfigureAwait(false)).MemberOfWorkshopId.HasValue)
+        {
+            var errorMessage = $"The main workshop (with ID = {dto.MemberOfWorkshopId}) for the workshop being created is a member of another workshop, so it cannot be the main workshop.";
+            throw new InvalidOperationException(errorMessage);
+        }
+
+        dto.AvailableSeats = dto.AvailableSeats.GetMaxValueIfNullOrZero();
+        Workshop createdWorkshop;
+
+        if (dto is WorkshopV2Dto v2Dto)
+        {
+            createdWorkshop = mapper.Map<Workshop>(v2Dto);
+        }
+        else
+        {
+            createdWorkshop = mapper.Map<Workshop>(dto);
+        }
+
+        createdWorkshop.Provider = await providerRepository.GetById(createdWorkshop.ProviderId).ConfigureAwait(false);
+        createdWorkshop.ProviderOwnership = createdWorkshop.Provider.Ownership;
+        createdWorkshop.ProviderTitle = createdWorkshop.Provider.FullTitle;
+        createdWorkshop.ProviderTitleEn = createdWorkshop.Provider.FullTitleEn;
+
+        if (!dto.Teachers.IsNullOrEmpty())
+        {
+            createdWorkshop.Teachers = dto.Teachers.Select(mapper.Map<Teacher>).ToList();
+        }
+
+        createdWorkshop.Tags = (await tagRepository.GetAll()).Where(tag => dto.TagIds.Contains(tag.Id)).ToList();
+        createdWorkshop.Status = WorkshopStatus.Open;
+
+        return createdWorkshop;
+    }
+
+    private async Task SetIdsToDefaultValue(WorkshopCreateUpdateDto dto)
```

Comment:
We create a new Workshop from WorkshopCreateUpdateDto which we get from frontend. It may have contain invalid Id value in passed Dto for properties - Address, DefaultTeacher, MemberOfWorkshop, WorkshopDescriptionItems, Teachers, etc...